link/src/main/java/com/meshconnect/link/ui/LinkActivity.kt

```diff
@@ -78,6 +84,10 @@ internal class LinkActivity : AppCompatActivity() {
     private val binding by viewBinding(LinkActivityBinding::inflate)
     private val viewModel by viewModel<LinkViewModel>(LinkViewModel.Factory())
 
+    private var isDarkTheme = false
+    private var themeColorTop: Int = 0
+    private var themeColorBottom: Int = 0
```

Comment:
Suggest having one function `applyTheme` to avoid using the global variables. This helps to keep all theme-related staff in one place. In that function, you obtain the theme property from the link token and update the UI accordingly.